genesiscommunitysuccess/custom-elements-lsp
# fix!: manifset analyzer debug script FUI-1286

⚠️ BREAKING CHANGE: changing the output bundle breaks the import path for cep plugins ⚠️

🤔  &nbsp; **What does this PR do?**

`@genesiscommunitysuccess/custom-elements-lsp`
- ⚠️ BREAKING CHANGE: Fixes the directory structure so source files are no longer import bundle files
- Convert analyzer debug script to typescript
- FIX: anlayzer debug script input argument no longer uses quotes, as this was breaking on windows powershell

Other
- Seperate out running the `lsp:analyze` debug step from the build to a seperate github actions step

📑  &nbsp; **How should this be tested?**

Update test steps to match your PR:

```
1. Checkout branch
2. `npm run bootstrap`
3. `npm run test`

Other packages are completely tested via tests from `npm run test`. To test the LSP is working correctly in VSCode:
1. `cd packages/showcase/example`
2. `code .` (or manually open VSCode via the UI to the `example` directory on your filesystem).
3. Open a typescript file such as `root.ts`
4. In the Command Palette choose `TypeScript: select typescript version...` and then choose the workspace version
5. The LSP should be enabled. You will need to make a code change after the LSP is enabled before you see any Intellisense.
```

1. All lsp functionality should work as normal
2. In the `example` directory from step one, check that `npm run lsp:analyze` runs, and that the final line looks something like this 
```
[log] Found dependency node_modules/@genesiscommunitysuccess/example-lib/dist/custom-elements.json in node_modules/@genesiscommunitysuccess/example-lib/dist
```

✅  &nbsp; **Checklist**



- [ ] I have added tests for my changes. - N/A
